src/input_common/gcadapter/gc_adapter.cpp

```diff
@@ -0,0 +1,371 @@
+// Copyright 2014 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include <chrono>
+#include <thread>
+#include "common/logging/log.h"
+#include "input_common/gcadapter/gc_adapter.h"
+
+namespace GCAdapter {
+
+Adapter::Adapter() {
+    if (usb_adapter_handle != nullptr) {
+        return;
+    }
+    LOG_INFO(Input, "GC Adapter Initialization started");
+
+    current_status = NO_ADAPTER_DETECTED;
+    libusb_init(&libusb_ctx);
+
+    StartScanThread();
+}
+
+GCPadStatus Adapter::CheckStatus(int port, const std::array<u8, 37>& adapter_payload) {
+    GCPadStatus pad = {};
+    bool get_origin = false;
+
+    ControllerTypes type = ControllerTypes(adapter_payload[1 + (9 * port)] >> 4);
+    if (type != ControllerTypes::None)
+        get_origin = true;
```

Comment:
```suggestion
    if (type != ControllerTypes::None) {
        get_origin = true;
    }
```

(formatting might be broken since I'm doing it by hand)